src/modules/cmdpal/ext/Microsoft.CmdPal.Ext.Indexer/Pages/ActionsListContextItem.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading.Tasks;
+using Microsoft.CmdPal.Ext.Indexer.Commands;
+using Microsoft.CmdPal.Ext.Indexer.Data;
+using Microsoft.CmdPal.Ext.Indexer.Properties;
+using Microsoft.CommandPalette.Extensions.Toolkit;
+using Windows.AI.Actions;
+using Windows.System;
+
+namespace Microsoft.CmdPal.Ext.Indexer.Pages;
+
+internal sealed partial class ActionsListContextItem : CommandContextItem
+{
+    private readonly string fullPath;
+    private readonly List<CommandContextItem> actions = [];
+    private static ActionRuntime actionRuntime;
+
+    public ActionsListContextItem(string fullPath)
+        : base(new NoOpCommand())
+    {
+        Title = Resources.Indexer_Command_Actions;
+        Icon = Icons.Actions;
+        RequestedShortcut = KeyChordHelpers.FromModifiers(alt: true, vkey: VirtualKey.A);
+        this.fullPath = fullPath;
+        UpdateMoreCommands();
+    }
+
+    public bool AnyActions() => actions.Count != 0;
+
+    private void ActionCatalog_Changed(global::Windows.AI.Actions.Hosting.ActionCatalog sender, object args)
+    {
+        UpdateMoreCommands();
+    }
+
+    private void UpdateMoreCommands()
+    {
+        try
+        {
+            if (actionRuntime == null)
```

Comment:
Do we need to lock this, or otherwise make sure we only initialize it once? I hit `CreateActionRuntime` like 25 times (once for each file result on the first search)